dali/operators/image/distortion/jpeg_compression_distortion_op.h

```diff
@@ -42,10 +42,10 @@ class JpegCompressionDistortion : public Operator<Backend> {
     const auto &input = ws.template Input<Backend>(0);
     output_desc.resize(1);
     const auto &in_sh = input.shape();
-    assert(in_sh.sample_dim() == 3);  // should be check by the layout
+    assert(in_sh.sample_dim() == 3 || in_sh.sample_dim() == 4);  // should be check by the layout
     for (int s = 0; s < in_sh.num_samples(); s++) {
-      DALI_ENFORCE(in_sh.tensor_shape_span(s)[2] == 3,
-        make_string("Expected RGB samples with HWC layout, got shape: ", in_sh[s]));
+      DALI_ENFORCE(in_sh.tensor_shape_span(s).back() == 3,
+                   make_string("Expected RGB samples with (F)HWC layout, got shape: ", in_sh[s]));
```

Comment:
```suggestion
                   make_string("Invalid number of channels. Expected a channel-last layout, with 3 channels (RGB), got shape: ", in_sh[s]));
```
Just a suggestion.

Reply:
1. Do we really require it to be RGB? I guess you _can_ have a monochrome JPEG, can't you? If we don't we should :)
2. I think that "3 channels (RGB)" is a bit redunant - our error messages are already quite long.

Reply:
Current implementation expects 3 channels, AFAIK

Reply:
Feel free to dismiss this suggestion if you don't find it better